cips/cip-blob-verified-signer.md

```diff
@@ -0,0 +1,92 @@
+- For the PFBs of the namespaces that the rollup is subscribed to, verify the `signer` matches the sequencer.
+- Use the share indexes of the `IndexWrapper` of the PFB to retrieve the blobs that match the PFB. Verify the blobs inclusion and finally process the blobs.
+  
+For rollups, using ZK, such as the case with Soverign, the flow is as follows:
+
+- Query all the blobs from the rollup namespace via RPC
+- For each blob, reconstruct the blob commitment.
+- Fetch the PFB namespace
+- Parse the PFB namespace and create a mapping from blob commitment -> PFB
+- (In zero-knowledge) Accept the list of all rollup blobs and the list of relevant PFBs as inputs
+- (In zero-knowledge) Verify that the claimed list of blobs matches the block header using the namespaced merkle tree
+- (In zero-knowledge) For each blob, find the PFB with the matching commitment and check that the sender is correct.
+- (In zero-knowledge) For each relevant PFB, check that the bytes provided match the namespaced merkle tree
+
+This is currently a needlessly complicated flow and more computationally heavy at constructing proofs. This CIP proposes an easier path for rollups that opt for this fork-choice rule
+
+## Specification
+
+This CIP introduces a new blob type (known henceforth as a v2 blob given the share format version change):
+
+```proto
+message Blob {
+  bytes namespace_id = 1;
+  bytes data = 2;
+  uint32 share_version = 3;
+  uint32 namespace_version = 4;
+  // new field
+  string signer = 5;
+}
+```
+
+Given proto's backwards compatibility, users could still submit the old blob type (in the `BlobTx` format) and signer would be processed as an empty string.
+
+The new block validity rule (In `PrepareProposal` and `ProcessProposal`) would thus be that if the signer was not empty, then it must match that of the PFB that paid for it. When validating the `BlobTx`, validators would check the equivalency of the PFB's `signer` to the Blob's `signer` (as well as verification of the signature itself).
+
+Although no version changes are required for protobuf encoded blobs, share encoding would change. Blobs containing a non empty signer string would be encoded using the new v2 format:
+
+![Diagram of V2 Share Format](../assets/cip-blob-verified-signer/blob-v2-share-format.svg)
+
+Blobs with an empty `signer` string would remain encoded using the v1 format. Note that in this diagram it is the `Info Byte` that contains the share version. Not to be confused with the namespace version.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+## Rationale
+
+Given the current specification change, the new loop is simplified:
+
+- Retrieve all blobs in the subscribed namespaces
+- Verify the namespaces inclusion
```

Comment:
[nit] can this be clarified b/c I'm not sure what it means. Does this step mean:

- Verify that the blobs from the previous step are present in the subscribed namespace?
- Verify that the blobs from the previous step are present and complete in the subscribed namespace? In other words, there are no other blobs in the subscribed namespace.
- Verify the blobs from the previous step are included in an NMT with root `RowRoot`? Additionally verify a `RowRoot` is included in some `DataRoot`?

Reply:
I would assume that inclusion and exclusion are done simultaneously i.e. we are proving that all the blobs we have are present in that namespace and there are no blobs in that namespace that we don't have.

Does that make sense? And would you like me to elaborate more in this document

Reply:
That sounds like the second option:

> Verify that the blobs from the previous step are present and complete in the subscribed namespace? In other words, there are no other blobs in the subscribed namespace.

But I think this would be clearer than what I previously wrote:

```
1. Retrieve all blobs in the subscribed namespace
2. Verify that the blobs from the previous step belong to the subscribed namespace and the list is complete (i.e. there are no blobs present in the subscribed namespace that are absent from the list of retrieved blobs).
```

Reply:
Resolved: https://github.com/celestiaorg/CIPs/pull/147/commits/62e5ef1ac8ffb785e5777cd837439d24b8a201f6